runtime-decompiler/src/main/java/org/jrd/backend/communication/FsAgent.java

```diff
@@ -131,25 +131,18 @@ private T operatetOnCp(String clazz, CpOperator<T> op) throws IOException {
                         if (clazz == null) {
                             throw new IOException("Trying to find class but no class is specified");
                         }
-                        ArchiveManager am = new ArchiveManager(c);
+                        ArchiveManager am = ArchiveManager.getInstance();
+                        am.setFile(c);
```

Comment:
This seems to be fundamentla wrong. You have singleton instance, and then you are usinf setFile on it, where the File c is the fundamental part of the arcivemanager. 

Maybe the arcvhive manager should remain instantiatable as it was, jsut with private cosntructor,  some ArchvieManagerSet would containe map<file, archivemanager>  and then it will have getisntanceForFile - which will return existing instance or create new if that file do nto yet have instance?